includes/optional-modules/class-rss.php

```diff
@@ -1141,6 +1339,67 @@ private static function is_republication_tracker_plugin_active() {
 		return class_exists( 'Republication_Tracker_Tool' );
 	}
 
+	/**
+	 * Handle AJAX search for taxonomy terms.
+	 */
+	public static function ajax_search_terms() {
+		check_ajax_referer( 'newspack_rss_search_terms', 'nonce' );
+
+		if ( ! current_user_can( 'edit_posts' ) ) {
+			wp_die( 'Insufficient permissions' );
+		}
+
+		if ( ! isset( $_POST['taxonomy'] ) || ! isset( $_POST['search'] ) ) {
+			wp_die( 'Invalid request' );
+		}
+
+		$taxonomy = sanitize_text_field( $_POST['taxonomy'] );
+		$search   = sanitize_text_field( $_POST['search'] );
+
+		if ( ! taxonomy_exists( $taxonomy ) ) {
+			wp_die( 'Invalid taxonomy' );
+		}
+
+		$terms = get_terms(
+			[
+				'taxonomy'   => $taxonomy,
+				'search'     => $search,
+				'hide_empty' => false,
+				'number'     => 50,
+				'fields'     => 'id=>name',
+			]
+		);
+
+		$results = [];
+		if ( ! is_wp_error( $terms ) ) {
+			foreach ( $terms as $term_id => $term_name ) {
+				$results[] = [
+					'id'   => $term_id,
+					'text' => $term_name,
+				];
+			}
+		}
+
+		wp_send_json( $results );
+	}
+
+	/**
+	 * Get custom taxonomies registered for posts (excluding built-in taxonomies).
+	 *
+	 * @return array Array of custom taxonomy names registered for posts.
+	 */
+	private static function get_custom_taxonomies_for_posts() {
+		$custom_taxonomies = get_taxonomies(
+			[
+				'object_type' => [ 'post' ],
+				'public'      => true,
+				'_builtin'    => false,
```

Comment:
I don't know if I'm misunderstanding how this works, but I don't see [the Brand taxonomy from Newspack Multibranded](https://github.com/Automattic/newspack-multibranded-site/blob/ac2076d9cad38353bb67ba72a19d43d67d826016/includes/class-taxonomy.php#L88) nor [the Sponsors taxonomy from Newspack Sponsors](https://github.com/Automattic/newspack-sponsors/blob/438f04eca64a6c54bdacdd8321d14e2624e431f0/includes/class-core.php#L316) when editing an RSS feed. Both are registered as public as far as I can tell.